col/src/main/java/vct/col/typerules/CoercingRewriter.scala

```diff
@@ -1301,4 +1407,260 @@ abstract class CoercingRewriter[Pre <: Generation]() extends Rewriter[Pre] with
         ParBlock(decl, iters, res(context_everywhere), res(requires), res(ensures), content)(region.blame)
     }
   }
+
+  def coerce(node: Verification[Pre]): Verification[Pre] = {
+    implicit val o: Origin = node.o
+    val Verification(tasks, expectedErrors) = node
+    Verification(tasks, expectedErrors)
+  }
+
+  def coerce(node: VerificationContext[Pre]): VerificationContext[Pre] = {
+    implicit val o: Origin = node.o
+    val VerificationContext(program) = node
+    VerificationContext(program)
+  }
+
+  def coerce(node: Program[Pre]): Program[Pre] = {
+    implicit val o: Origin = node.o
+    val Program(decl) = node
+    Program(decl)(node.blame)
+  }
+
+  def coerce(node: Type[Pre]): Type[Pre] = {
+    implicit val o: Origin = node.o
+    node match {
+      case value: TNotAValue[_] =>
+        value
+      case TUnion(types) =>
+        TUnion(types)
+      case TArray(element) =>
+        TArray(element)
+      case TPointer(element) =>
+        TPointer(element)
+      case TType(t) =>
+        TType(t)
+      case TVar(ref) =>
+        TVar(ref)
+      case compositeType: CompositeType[_] =>
+        compositeType
+      case primitiveType: PrimitiveType[_] =>
+        primitiveType
+      case declaredType: DeclaredType[_] =>
+        declaredType
+      case cType: CType[_] =>
+        cType
+      case javaType: JavaType[_] =>
+        javaType
+      case lType: PVLType[_] =>
+        lType
+      case silverType: SilverType[_] =>
+        silverType
+    }
+  }
+
+  def coerce(node: LoopContract[Pre]): LoopContract[Pre] = {
+    implicit val o: Origin = node.o
+    node match {
+      case li @ LoopInvariant(invariant) =>
+        LoopInvariant(res(invariant))(li.blame)
+      case ic @ IterationContract(requires, ensures, context_everywhere) =>
+        IterationContract(res(requires), res(ensures), res(context_everywhere))(ic.blame)
+    }
+  }
+
+  def coerce(node: CatchClause[Pre]): CatchClause[Pre] = {
+    implicit val o: Origin = node.o
+    val CatchClause(decl, body) = node
+    CatchClause(decl, body)
+  }
+
+  def coerce(node: IterVariable[Pre]): IterVariable[Pre] = {
+    implicit val o: Origin = node.o
+    val IterVariable(variable, from, to) = node
+    IterVariable(variable, int(from), int(to))
+  }
+
+  def coerce(node: SignalsClause[Pre]): SignalsClause[Pre] = {
+    implicit val o: Origin = node.o
+    val SignalsClause(binding, assn) = node
+    SignalsClause(binding, res(assn))
+  }
+
+  def coerce(node: DecreasesClause[Pre]): DecreasesClause[Pre] = {
+    implicit val o: Origin = node.o
+    node match {
+      case DecreasesClauseAssume() =>
+        DecreasesClauseAssume()
+      case DecreasesClauseNoRecursion() =>
+        DecreasesClauseNoRecursion()
+      case DecreasesClauseTuple(exprs) =>
+        DecreasesClauseTuple(exprs)
```

Comment:
I changed my mind about this: since we currently only support integers, lets coerce the expressions to `int`.